src/control/PlanarArmController.h

```diff
@@ -42,7 +47,27 @@ template <unsigned int N> class PlanarArmController {
 	void set_setpoint(const navtypes::Vectord<N>& targetJointPos) {
 		Eigen::Vector2d newSetPoint = kin.jointPosToEEPos(targetJointPos);
 		std::lock_guard<std::mutex> lock(mutex);
-		setpoint = newSetPoint;
+		double radius = kin.getSegLens().sum() * safetyFactor;
+		setpoint = normalizeVectorWithinRadius(newSetPoint, radius);
+	}
+
+	/**
+	 * @brief Normalize the input vector to have a set radius,
+	 *  	  while maintaining the same direction it did before if it exceeds that set radius.
+	 *
+	 * @param input The input vector to normalize.
+	 * @param radius The radius to normalize the vector to.
+	 */
+	Eigen::Vector2d normalizeVectorWithinRadius(Eigen::Vector2d input, double radius) {
```

Comment:
Also we should rename `input` to something more useful, like `eePos` or something, and rename `normalizeVectorWithinRadius` to something more indicative as well